src/lib/stitching/exchange/link.ts

```diff
@@ -33,7 +33,20 @@ export const createExchangeLink = () => {
     return { headers }
   })
 
+  const analyticsMiddleware = setContext((_request, context) => {
+    const locals = context.graphqlContext && context.graphqlContext.res.locals
+    if (!locals) return context.graphqlContext
+    return {
+      ...context.graphqlContext,
+      headers: {
+        ...context.graphqlContext.headers,
+        "x-origin-user-agent": locals.userAgent,
```

Comment:
Is it a customary practice to name the header like so? I thought a header like this was usually forwarded as-is (i.e. `User-Agent`), but actually not sure now and can’t easily google a great answer.

Reply:
I think if you're adding custom headers the standard practice is to prefix them with 'x-', but not sure what the standard practice is around forwarding headers. i.e. whether you're supposed to overwrite the forwarded header (essentially spoofing) or put it in a custom header. It probably just depends on the use case. I did this because it seemed like the safer option, but totally happy to change it?

Reply:
Should we actually just send the user's UA string _as_ user agent? I'm thinking that server strings like `Metaphysics`, or `node server` (which you see tons of when you have these apps making requests back and forth) aren't actually useful. If the meaningful data is the end user's string (and all of these requests are on 'behalf' of a user or user action anyway), maybe we should just do that?

Alternatively, you can take the user agent string and append `; Metaphysics` to the end, so we know its 'forwarded' by MP.

We actually do that here: https://github.com/artsy/metaphysics/blob/9b472f90c97e9568fe331808eb9ee05ede5bb6b5/src/lib/apis/fetch.ts#L20-L24

Reply:
Interesting problem! I'm sure, technically speaking, it's not compliant for Metaphysics to overwrite the standard `User-Agent` header. In that sense implementing a custom header is more appropriate. I _do_ see some mentions of `X-Device-User-Agent` or `X-Original-User-Agent` on the internet, so we might want to use those rather than choosing a new name for this purpose.

On the other hand, using the standard header allows it to appear in logs, etc. without further customization. I guess it's a question of whether Metaphysics is just a "proxy" or not.

One thought: If we'll use a custom header for this, it should propagate the "original" value reliably from upstream clients. I.e., rather than simply supplying this request's `User-Agent` value, it should give an existing `X-...-User-Agent` value precedence. Force's server-side could provide this to MP in the future.

Reply:
I went with overwriting the `User-Agent` header for now. Seems like the simplest thing to do in this situation. @joeyAghion what do you mean by "an existing `X-...-User-Agent` value precedence"?